### PR TITLE
fix: specifies the max length for crate name

### PR DESCRIPTION
**Pull request by EstebanBorai**
Provides the maximum length on the crate name as is defined
in the [crate's validation source code from crates.io][1].

[1]: https://github.com/rust-lang/crates.io/blob/3fc08ba57e782cff422a2f92503f09287d1a5140/src/models/krate.rs#L74

### CODE REVIEW

**Comment by ehuss** [comment]
Thanks for the PR!  Since this is a crates.io restriction, can you move that to the end of the paragraph where those additional restrictions are listed?

**Comment by EstebanBorai** [comment]
> Thanks for the PR! Since this is a crates.io restriction, can you move that to the end of the paragraph where those additional restrictions are listed?

Of course!

**Comment by EstebanBorai** [comment]
@ehuss thanks for the feedback, given that the list is growing, I changed the format to list so it's easier to read, wdyt?

File: src/doc/src/reference/manifest.md [hunk 1]
```diff
@@ -73,12 +73,17 @@ The package name is an identifier used to refer to the package. It is used
 when listed as a dependency in another package, and as the default name of
 inferred lib and bin targets.
 
-The name must use only [alphanumeric] characters or `-` or `_`, and cannot be empty.
+The name must use only [alphanumeric] characters or `-` or `_`, and cannot be
+empty.
```
Comment (weihanglo) [inline src/doc/src/reference/manifest.md:77]: (not a blocker)
Could we not wrap this? ehuss prefers to [semantic linefeeds](https://github.com/rust-lang/cargo/pull/10916#discussion_r934132791) which I bought into as well 😆

File: src/doc/src/reference/manifest.md [hunk 2]
```diff
@@ -73,12 +73,17 @@ The package name is an identifier used to refer to the package. It is used
...
 the package name, such as enforcing that it is a valid Rust identifier and not
-a keyword. [crates.io] imposes even more restrictions, such as
-enforcing only ASCII characters, not a reserved name, not a special Windows
-name such as "nul", is not too long, etc.
+a keyword. [crates.io] imposes even more restrictions, such as:
+
+- Only ASCII characters are allowed
+- Do not use reserved names
+- Do not use special Windows name such as "nul"
+- Use a maximum of 64 characters of length
```
Comment (weihanglo) [inline src/doc/src/reference/manifest.md:86]: Adding periods looks nicer, I guess?

```suggestion
- Only ASCII characters are allowed.
- Do not use reserved names.
- Do not use special Windows name such as "nul".
- Use a maximum of 64 characters of length.
```

**Review by weihanglo — APPROVED** [review]

File: src/doc/src/reference/manifest.md (hunk 2 above)
Comment (EstebanBorai) [reply src/doc/src/reference/manifest.md]: Yeah! I agree!

File: src/doc/src/reference/manifest.md (hunk 1 above)
Comment (EstebanBorai) [reply src/doc/src/reference/manifest.md:77]: Of course!

**Comment by weihanglo** [comment]
Thank you!

@bors r+

**Comment by ehuss** [comment]
@bors retry

MSVC stable failed running `benches/resolve.rs`, which ran the resolver on the cargo workspace, and failed with:

```
Testing resolve_ws/cargo
    Updating `local-snapshot` index
 Downloading crates ...
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: failed to download from `[https://crates.io/api/v1/crates/winapi/0.3.9/download`](https://crates.io/api/v1/crates/winapi/0.3.9/download%60)

Caused by:
    [1] Unsupported protocol (Received HTTP/0.9 when not allowed)', benches\resolve.rs:45:6
```

I have never seen that error before. Very strange!

**Comment by weihanglo** [comment]
@bors retry

The curl bug haven't been fixed yet, but `curl-sys 0.4.57+curl-7.85.0` is yanked for the time being. I'll give it another shot.

ref: https://github.com/alexcrichton/curl-rust/issues/460#issuecomment-1239985109

**Pull request merged**